src/foam/nanos/rope/ROPEUtils.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.rope',
+  name: 'NextRelationshipsList',
+
+  properties: [
+    {
+      name: 'nextRelationships',
+      class: 'List',
+      javaType: 'java.util.List<String>'
+    }
+  ]
+
+});
+
+foam.CLASS({
+  package: 'foam.nanos.rope',
+  name: 'RelationshipMap',
+
+  properties: [
+    {
+      name: 'map',
+      class: 'Map',
+      javaType: 'java.util.Map<String, NextRelationshipsList>'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'get',
+      javaType: 'java.util.List<String>',
+      args: [
+        { name: 'relationshipKey', class: 'String' }
+      ],
+      javaCode: `
+        return getMap().containsKey(relationshipKey) && getMap().get(relationshipKey) != null ? getMap().get(relationshipKey).getNextRelationships() : null;
+      `
+    },
+  ]
+
+});
+
+foam.CLASS({
+  package: 'foam.nanos.rope',
+  name: 'CRUDMap',
+
+  javaImports: [
+    'java.util.Map',
+    'java.util.List'
+  ],
+
+  properties: [
+    {
+      name: 'create',
+      class: 'Map',
+      javaType: 'java.util.Map<String, NextRelationshipsList>'
+    },
+    {
+      name: 'read',
+      class: 'Map',
+      javaType: 'java.util.Map<String, NextRelationshipsList>'
+    },
+    {
+      name: 'update',
+      class: 'Map',
+      javaType: 'java.util.Map<String, NextRelationshipsList>',
+    },
+    {
+      name: 'delete',
+      class: 'Map',
+      javaType: 'java.util.Map<String, NextRelationshipsList>',
+    }
+  ],
+  
+  axioms: [
+    {
+      name: 'javaExtras',
```

Comment:
Any reason these methods are added via `javaExtras` rather than `methods`?

Reply:
yes, I couldn't think of a good name for the methods except get, so I tried to overload `get` with different arguments instead, but method overloading is not supported in foam

Reply:
Ok, that makes sense. Please include a comment explaining that in the code.

Reply:
ok, commented